### ISSUE TITLE
Define an MSRV policy for the crate

### ISSUE THREAD

**Issue by zslayton** [amazon-ion/ion-rust#800]
> while we still can

There's a [cfg_version](https://doc.rust-lang.org/nightly/unstable-book/language-features/cfg-version.html) RFC that will help with this.

Also, if we haven't already done so, we should define a MSRV policy for this crate.

_Originally posted by @popematt in https://github.com/amazon-ion/ion-rust/pull/796#discussion_r1672709415_

**Comment by zslayton** [comment]
I propose stealing [the MSRV policy from the `jiff` crate](https://github.com/BurntSushi/jiff?tab=readme-ov-file#minimum-rust-version-policy), as it seems reasonable to me:

> This crate's minimum supported rustc version is 1.70.0.
> 
> The policy is that the minimum Rust version required to use this crate can be increased in minor version updates. For example, if jiff 1.0 requires Rust 1.20.0, then jiff 1.0.z for all values of z will also require Rust 1.20.0 or newer. However, jiff 1.y for y > 0 may require a newer minimum version of Rust.

**Comment by popematt** [comment]
@zslayton, would it be fair to summarize that as "increasing the MSRV requires a new minor version"?

Another one that looks reasonable to me is [this proposal for libc](https://github.com/rust-lang/libs-team/issues/72).

**Comment by zslayton** [comment]
> would it be fair to summarize that as "increasing the MSRV requires a new minor version"?

Yes, without a "we support the last _N_ releases" requirement. I could get behind a "last N" policy for a sufficiently small `N`.
